polaris-react/src/components/IndexTable/components/Row/Row.tsx

```diff
@@ -39,6 +39,8 @@ export interface RowProps {
    * @default "Select {resourceName}"
    */
   accessibilityLabel?: string;
+  /** Whether the row should have border-top */
+  borderless?: boolean;
```

Comment:
```suggestion
  border?: boolean;
```

Can we simplify the prop name to `border` with the default being `true`? When no border is desired, the usage would be `border={false}`. Using `border` is more aligned with the CSS semantic.

Sorry, I know that will also affect the class names and stories 😅

Reply:
After thinking more about it, using a string would be more flexible for future needs. Could we use `border="none"`?

Reply:
Thank you for your feedback, @sam-b-rose ! 
In case with `borderless` we add `TableRow-borderless` class and remove borders with css.
If you suggest  `border?: boolean;` or `border="none"`, how this should be styled? Should we by default add `TableRow-border` class?
With `border="none"` what other values it should have, `border="yes"`?
Thank you!

Reply:
Hey @oksanashopify! Sam is suggesting string typing so that if there is a need in the future for other variants of border styling, they can simply be unioned with "none" as needed (like `"none" | "some other option"`). For the classname, you can update to follow style variant naming convention `TableRow-borderNone`.

Reply:
@sam-b-rose @oksanashopify what do you think of aligning the prop with `Box`?

```tsx
/** The border-top color of the row
  * @default 'secondary'
  */
borderColor?: 'secondary' | 'transparent';
```

(IndexTable.css @ line 158)
```css
.TableRow {
  background-color: var(--p-color-bg-surface);
  cursor: pointer;
  border-width: var(--p-border-width-025);
  border-style: solid;

  &.TableRow-unclickable {
    cursor: auto;
  }

  &:first-child {
    border-color: var(--p-color-border);
  }

  &.borderColorSecondary {
    border-color: var(--p-color-border-secondary);
  }

  &.borderColorTransparent {
    border-color: transparent;
  }

```